doozer/doozerlib/util.py

```diff
@@ -561,27 +563,101 @@ def oc_image_info__caching(pull_spec: str, go_arch: str = 'amd64') -> Dict:
 
 
 @retry(reraise=True, wait=wait_fixed(3), stop=stop_after_attempt(3))
-async def oc_image_info_async(pull_spec: str, go_arch: str = 'amd64') -> Dict:
+async def oc_image_info_async(
+        pull_spec: str,
+        go_arch: str = 'amd64',
+        registry_config: Optional[str] = None,
+) -> Dict:
     """
     Returns a Dict of the parsed JSON output of `oc image info` for the specified
     pullspec. Filter by os because images can be multi-arch manifest lists
     (which cause oc image info to throw an error if not filtered).
 
-    Use oc_image_info__caching if you think the image won't change during the course of doozer's execution.
+    Use oc_image_info_async__caching if you think the image won't change during the course of doozer's execution.
+
+    :param pull_spec: The image pullspec to query.
+    :param go_arch: The Go architecture to filter by.
+    :param registry_config: The path to the registry config file. If not provided, the default registry config will be used.
+    :param registry_username: The username to authenticate with the registry.
+    :param registry_password: The password to authenticate with the registry.
+    :return: The parsed JSON output of `oc image info`.
     """
-    cmd = ['oc', 'image', 'info', f'--filter-by-os={go_arch}', '-o', 'json', pull_spec]
+    options = [f'--filter-by-os={go_arch}', '-o', 'json']
+    if registry_config:
+        options.extend([f'--registry-config={registry_config}'])
+    cmd = ['oc', 'image', 'info'] + options + [pull_spec]
     _, out, _ = await exectools.cmd_gather_async(cmd)
     return json.loads(out)
 
 
+async def oc_image_info_with_auth_async(
```

Comment:
it looks like we could just define a single `oc_image_info` function that accepts optional auth params instead of having `oc_image_info_with_auth_async` fall back to `oc_image_info_async` in case username and password are not provided?

Reply:
That is possible. I was just lazy to do that. Let me check how easy or hard to change that.

Reply:
Updated